services/gitlab/gitlab-top-language.service.js

```diff
@@ -0,0 +1,80 @@
+import Joi from 'joi'
+import { optionalUrl } from '../validators.js'
+import { pathParam, queryParam } from '../index.js'
+import GitLabBase from './gitlab-base.js'
+import { description, httpErrorsFor } from './gitlab-helper.js'
+
+const schema = Joi.object()
+  .pattern(Joi.string(), Joi.number().min(0).max(100).precision(2))
+  .required()
```

Comment:
The object key and value should both be `.required()`, so

```suggestion
const schema = Joi.object()
  .pattern(
    Joi.string().required(),
    Joi.number().min(0).max(100).precision(2).required(),
  )
  .required()
```

here